### ISSUE TITLE
IPTV Archive Downloader records at wrong time

### ISSUE THREAD

**Issue by stuckinthe** [tbrek/tbrek-kodi-repo#21]
I am running a RPI3+ with 20.2 on Xbian. I am having an issue with the recording times as it starts the record as the end time and then records for the length of the show. So if the show is at 12:00 - 12:30, it will record from 12:30 to 1:00 which is naturally the wrong show. Also, I am also confused on what the best setting is for service (period, time, none).

I have checked my date on my OS and the date on Kodi and they are correct. Does anyone know how I can fix this?

**Comment by tbrek** [comment]
Who is your IPTV provider? What EPG do you use. What are your EPG shift settings?

**Comment by stuckinthe** [comment]
Thanks for the help. I don't have a specific provider. I have tried using the IPTV Simple option and then I also tried a m3u file. I didn't see an option for the EPG so I wasn't sure if it reads it in from the m3u as I don't have the epg set in my m3u since I use IPTV merge. I have set my shift to zero. I have tried -1 and 1 but this didn't work. I wasn't sure what to put for the archive format.

I notice when the plugin builds the database I get an error that says could not download live stream. Also, I looked at the queue file and it shows the proper start and stop time.

**Comment by tbrek** [comment]
This plugin mainly support two providers Tele and PlusX + custom ones. Without knowing what Archive format your IPTV provider uses or without EPG provided it's difficult to say what the problem might be. 

When it's set to SimpleIPTV it will use IPTV m3u defined there as well as EPG.

**Comment by stuckinthe** [comment]
I tried using SimpleIPTV and it selects the program and date and time properly per the queue file. It records the show properly in the ts format albeit records the wrong time. I uncommented the log statements in main.py and then ran kodi in debug mode. UTC shows a half an hour later and in turn LUTC shows a half an hour later.  I am not sure why utc conversion is off by 30 minutes. I haven't tried an hour show to see if it's off by an hour.

**Comment by electro575** [comment]
How have you arrived to install and work addon iptv recorder and use archive downloader with Kodi 20.2 ?
Thanks a lot

**Comment by stuckinthe** [comment]
Are you having the same problem with it recording at the wrong time or it won't install. I edited the code to fix the issue of recording the wrong time but I have since moved on to 21 and I haven't tried installing.

**Comment by electro575** [comment]
With the repo installer of kodi 20.2, i must add lines on file addon.xml to install the tbrek repositorie.

```
        <dir maxversion="20.2.0">
            <info compressed="false">https://tbrek.github.io/tbrek-kodi-repo/matrix/addons.xml</info>
            <checksum>https://tbrek.github.io/tbrek-kodi-repo/matrix/addons.xml.md5</checksum>
            <datadir zip="true">https://tbrek.github.io/tbrek-kodi-repo/matrix/datadir/</datadir>
        </dir>
```

But i supposed it's not the right lines to write.

I had the possibility to install "plugin.video.iptv.archive.downloader-0.7.6.zip" from the repositorie.

I don't have the possibility to record a channel !

And when i go to iptv archive downloader extension, the log is the following.

```
2023-12-28 16:54:53.037 T:1743    error <general>: Repository add-on repository.tbrek.tbrek-kodi-repo does not have any directory and won't be able to update/serve addons! Please fix the addon.xml definition
2023-12-28 16:55:01.563 T:1060     info <general>: Loading skin file: MyVideoNav.xml, load type: KEEP_IN_MEMORY
2023-12-28 16:55:02.397 T:1847     info <general>: [xbmcswift2] Request for "/" matches rule for function "index"
2023-12-28 16:55:15.451 T:1849     info <general>: [xbmcswift2] Request for "/full_service" matches rule for function "full_service"
2023-12-28 16:55:15.452 T:1849     info <general>: '-------------------- Queue -------------------'
2023-12-28 16:55:15.452 T:1849     info <general>: '---------------- End of queue ----------------'
2023-12-28 16:55:16.527 T:1849     info <general>: '-------------------- Queue -------------------'
2023-12-28 16:55:16.528 T:1849     info <general>: '---------------- End of queue ----------------'
2023-12-28 16:55:16.530 T:1849     info <general>: CPythonInvoker(8, /storage/.kodi/addons/plugin.video.iptv.archive.downloader/main.py): waiting on thread 3415192192
2023-12-28 16:55:17.371 T:1852     info <general>: [xbmcswift2] Request for "/" matches rule for function "index"
```

And at each time i try to use a menu of this add-on, i have this error :

`error <general>: Repository add-on repository.tbrek.tbrek-kodi-repo does not have any directory and won't be able to update/serve addons! Please fix the addon.xml definition `

It's because i use Libreelec OS ? I suppose not, i use kodi 20.2.0.

The following add-on plugin.video.iptv.recorder of this URL project is written in python 2 instead of kodi 20.2 use python 3.

https://github.com/primaeval/repository.primaeval

With this add-on, i have access to a menu to see channel from m3u file, but after correcting same problem of python 2, i have problem with SQL part :/

I'm going to holiday, i come back into 1 week. 

Thanks for your feedback.

**Comment by stuckinthe** [comment]
I downloaded the zip file and installed the plugin this way without using the repo.

plugin.video.iptv.recorder is different than iptv.downloader. There is a version of plugin.video.iptv.recorde that installs on Kodi 20.2. You will need to find the link on the kodi forums. It worked for me but crashed after recording. I think it had a write to SQL issue but I couldn't figure it out.

**Comment by tbrek** [comment]
Please start a new issue with following details:

1. Kodi Version
2. Partial log stating the error.

In mean time I'll see If I can make this plugin Kodi 20 compatible.
